pallets/tx_fee/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::weights::{DispatchInfo, GetDispatchInfo};
+use pallet_transaction_payment::{FeeDetails, OnChargeTransaction};
+use sp_runtime::{traits::Dispatchable, FixedPointOperand};
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+// Type aliases used for interaction with `OnChargeTransaction`.
+pub(crate) type OnChargeTransactionOf<T> =
+	<T as pallet_transaction_payment::Config>::OnChargeTransaction;
+
+// Balance type alias.
+pub(crate) type BalanceOf<T> = <OnChargeTransactionOf<T> as OnChargeTransaction<T>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_transaction_payment::Config {}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	impl<T: Config> Pallet<T> {
+		pub fn compute_extrinsic_cost<Extrinsic: sp_runtime::traits::Extrinsic + GetDispatchInfo>(
```

Comment:
These are RPC endpoints which should exist in their corresponding pallet. Why are we taking these out?

Reply:
Can you please point me to where you got this pattern from?

Reply:
Not really! then you would introduce dependencies of transaction pallets in each one of those, ideally I would handle all such translations at one spot, especially knowing this will evolve into complexities of doing some math around fee adjustments for specific cases etc.

Otherwise, I will have all pallets coupled with transaction pallet and do its own thing.

Reply:
Further, we can have traits or say enums defined where logic can be updated around say Schema, or Message Cost etc.

Reply:
I guess I dont understand what special transactions we are doing?

Reply:
A schema could have different fee based on its retention policy for example or a message storage cost could differ based on its storage period, and so on

Reply:
We don't have this flushed out yet. We are already well aware that we do not know the details of our retention policy. Lets wait for when we have this problem.

Reply:
I am just going off of discussion but can always hold off though you confused me with "we may not have that need" which is again may or may not question because this is not trying to solve it but put a placeholder because I am sure we have that requirement.